### PR TITLE
Implemented save logic in create quiz

### PR DESCRIPTION
**Pull request by MariaAfonina**
https://github.com/ita-social-projects/SpaceToStudy-Client/assets/104988390/a757a58e-7b42-41a6-a241-8c5d0f5c9c17

Screenshot 2023-11-10 at 12 06 50

Screenshot 2023-11-09 at 19 45 44

### CODE REVIEW

File: src/containers/my-quizzes/create-or-edit-quiz-container/CreateOrEditQuizContainer.constants.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  title: '',
+  description: '',
+  category: null,
+  items: [],
+  questionTitle: '',
+  text: '',
+  openAnswer: '',
+  answers: [],
+  questionCategory: null,
+  type: sortQuestions[0].value
```
Comment (Tolik170) [inline src/containers/my-quizzes/create-or-edit-quiz-container/CreateOrEditQuizContainer.constants.ts:13]: you don`t need all these fields, also, this object is redundant

File: src/containers/my-quizzes/create-or-edit-quiz-container/CreateOrEditQuizContainer.tsx [hunk 2]
```diff
@@ -1,37 +1,95 @@
...
+
+  const { fetchData: addNewQuiz } = useAxios<Quiz, NewQuiz>({
+    service: addQuiz,
+    fetchOnMount: false,
+    defaultResponse,
+    onResponse: handleResponse,
+    onResponseError: onResponseError
+  })
+
+  const { data } = useForm<NewQuiz>({
```
Comment (Tolik170) [inline src/containers/my-quizzes/create-or-edit-quiz-container/CreateOrEditQuizContainer.tsx:89]: you don`t need to use useForm hook here, you can just pass title, description and questions from props to addNewQuiz callback

Comment (MariaAfonina) [reply src/containers/my-quizzes/create-or-edit-quiz-container/CreateOrEditQuizContainer.tsx:89]: Yes, you are right, thanks!

**Review by TSlashDreamy — APPROVED** [review]

**Review by dmtrth25 — APPROVED** [review]

File: src/types/quizzes/interfaces/quizzes.interface.ts [hunk 3]
```diff
@@ -11,3 +17,9 @@ export interface Quiz extends CommonEntityFields {
   author: Pick<UserResponse, '_id'>
   category: Category | null
 }
+
+export interface NewQuiz {
```
Comment (Tolik170) [inline src/types/quizzes/interfaces/quizzes.interface.ts:21]: better name it as CreateQuizParams

**Review by Tolik170 — APPROVED** [review]

**Review by abalanovsky — APPROVED** [review]

**Review by maarekua — APPROVED** [review]

**Pull request merged**